src/com/dre/brewery/listeners/InventoryListener.java

```diff
@@ -1,25 +1,177 @@
+	private HashMap<Block, Integer> trackedBrewers = new HashMap<Block, Integer>();
+	
+	/**
+	 * Start tracking distillation for a person when they open the brewer window.
+	 * @param event
+	 */
+	@EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
+	public void onBrewerOpen(InventoryOpenEvent event) {
+		if (!P.use1_9) return;
+		HumanEntity player = event.getPlayer();
+		Inventory inv = event.getInventory();
+		if (player == null || inv == null || !(inv instanceof BrewerInventory)) return;
+		
+		P.p.debugLog("Starting brew inventory tracking");
+		trackedBrewmen.add(player.getUniqueId());
+	}
+	
+	/**
+	 * Stop tracking distillation for a person when they close the brewer window.
+	 * @param event
+	 */
+	@EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
+	public void onBrewerClose(InventoryCloseEvent event) {
+		if (!P.use1_9) return;
+		HumanEntity player = event.getPlayer();
+		Inventory inv = event.getInventory();
+		if (player == null || inv == null || !(inv instanceof BrewerInventory)) return;
+		
+		P.p.debugLog("Stopping brew inventory tracking");
+		trackedBrewmen.remove(player.getUniqueId());
+	}
 
+	/**
+	 * Clicking can either start or stop the new brew distillation tracking.
+	 * Note that server restart will halt any ongoing brewing processes and
+	 * they will _not_ restart until a new click event.
+	 * 
+	 * @param event the Click event.
+	 */
+	@EventHandler(priority = EventPriority.HIGH, ignoreCancelled = true)
+	public void onBrewerClick(InventoryClickEvent event) {
+		if (!P.use1_9) return;
+		HumanEntity player = event.getWhoClicked();
+		Inventory inv = event.getInventory();
+		if (player == null || inv == null || !(inv instanceof BrewerInventory)) return;
+		
+		UUID puid = player.getUniqueId();
+		if (!trackedBrewmen.contains(puid)) return;
+		
+		if (InventoryType.BREWING != inv.getType()) return;
+		
+		BrewerInventory brewer = (BrewerInventory) inv;
+		final Block brewery = brewer.getHolder().getBlock();
+		
+		// If we were already tracking the brewer, cancel any ongoing event due to the click.
+		Integer curTask = trackedBrewers.get(brewery);
+		if (curTask != null) {
+			Bukkit.getScheduler().cancelTask(curTask); // cancel prior
+		}
+		
+		// Now check if we should bother to track it.
+		trackedBrewers.put(brewery, new BukkitRunnable() {
+			private int brewTime = 401;
+			@Override
+			public void run() {
+				BlockState now = brewery.getState();
+				if (now instanceof BrewingStand) {
+					BrewingStand stand = (BrewingStand) now;
+					// check if still custom
+					BrewerInventory brewer = stand.getInventory();
+					if (isCustomAndDistill(brewer) ) {
+						
+						// Still a valid brew distillation
+						brewTime = brewTime - 1; // count down.
+						stand.setBrewingTime(brewTime); // arbitrary for now
+						
+						if (brewTime <= 1) { // Done!
+							BrewEvent doBrew = new BrewEvent(brewery, brewer);
+							Bukkit.getServer().getPluginManager().callEvent(doBrew);
```

Comment:
Does this need to be called for the whole server? What if another plugin cancelled, or uncancelled it?